tests/phpunit/integration/Core/Modules/REST_Modules_ControllerTest.php

```diff
@@ -0,0 +1,633 @@
+<?php
+/**
+ * REST_Modules_ControllerTest
+ *
+ * @package   Google\Site_Kit
+ * @copyright 2022 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+namespace Google\Site_Kit\Tests\Core\Modules;
+
+use Google\Site_Kit\Context;
+use Google\Site_Kit\Core\Modules\Modules;
+use Google\Site_Kit\Core\REST_API\REST_Routes;
+use Google\Site_Kit\Core\Storage\Options;
+use Google\Site_Kit\Core\Storage\User_Options;
+use Google\Site_Kit\Tests\TestCase;
+use Google\Site_Kit\Tests\FakeHttpClient;
+use WP_REST_Request;
+
+class REST_Modules_ControllerTest extends TestCase {
+
+	public function set_up() {
+		parent::set_up();
+
+		$this->user         = $this->factory()->user->create_and_get( array( 'role' => 'administrator' ) );
+		$this->context      = new Context( GOOGLESITEKIT_PLUGIN_MAIN_FILE );
+		$this->options      = new Options( $this->context );
+		$this->user_options = new User_Options( $this->context, $this->user->ID );
+		$this->modules      = new Modules( $this->context, $this->options, $this->user_options );
```

Comment:
Apologies if this wasn't entirely clear before, but we want to be using `REST_Modules_Controller` directly here as much as possible. Assuming there isn't a reason we can't do this, we'd instantiate `REST_Modules_Controller` here as well and only register this one rather than via `Modules`. Does that make sense?

Reply:
Ah now i understand what you meant. This has been updated.